system/db/content.go

```diff
@@ -36,6 +36,98 @@ func SetContent(target string, data url.Values) (int, error) {
 	return update(ns, id, data)
 }
 
+// UpdateContent inserts or updates values in the database.
+// Updated content will be merged into existing values from the database.
+// The `target` argument is a string made up of namespace:id (string:int)
+func UpdateContent(target string, data url.Values) (int, error) {
+	t := strings.Split(target, ":")
+	ns, id := t[0], t[1]
+
+	// check if content id == -1 (indicating new post).
+	// if so, run an insert which will assign the next auto incremented int.
+	// this is done because boltdb begins its bucket auto increment value at 0,
+	// which is the zero-value of an int in the Item struct field for ID.
+	// this is a problem when the original first post (with auto ID = 0) gets
+	// overwritten by any new post, originally having no ID, defauting to 0.
+	if id == "-1" {
```

Comment:
I think we should remove this option for a `id=-1` check to then call `insert()`. It enables an outside client to bypass the `Externalable` interface, and add content, not just update it. 

Imagine a use-case where a CMS admin is putting content into their system, and wants other users to be able to only edit this content, but no one can add content except through the CMS. `Updateable` would need to be enabled, but this current implementation would still allow outsiders to add content, which I think should only be allowed when the type explicitly implements `Externalable`. 

What do you think..?